source/crd.go

```diff
@@ -117,7 +117,14 @@ func (cs *crdSource) Endpoints() ([]*endpoint.Endpoint, error) {
 	}
 
 	for _, dnsEndpoint := range result.Items {
-		endpoints = append(endpoints, dnsEndpoint.Spec.Endpoints...)
+		// Make sure that all endpoints have targets for A or CNAME type
+		for _, endPoint := range dnsEndpoint.Spec.Endpoints {
```

Comment:
Nit: please use endpoint instead of endPoint

Reply:
done, thanks for catching